robottelo/hosts.py

```diff
@@ -853,12 +855,19 @@ class Satellite(Capsule):
     def __init__(self, hostname=None, **kwargs):
         hostname = hostname or settings.server.hostname
         super().__init__(hostname=hostname, **kwargs)
-        self._init_nailgun()
-        self._init_cli()
-        self._init_airgun()
+        # create dummy classes for later population
+        self._api = type('api', (), {'_configured': False})
```

Comment:
This is bit wild for my taste. Is this some kind of Singletone or [Borg](https://www.oreilly.com/library/view/python-cookbook/0596001673/ch05s23.html) design pattern? Can you explain, perhaps in the comment in the code, why are there both -- lines 859 and 867? Is this because you want to have the `Satellite._api` available as class attribute as well as an instance attribute `self._api`?

Type is creating a class. Are you willing To make the name CamelCased? `type('API', ...` or `type('Api', ...` please?

Reply:
these are just handy objects that play slightly nicer, than some other alternatives, with what we need to do for pickle.

both are instance attributes only. The addition of 867 etc is extra redundancy post-pickle since __init__ is not called when an object is brought back from a pickled state. also see above regarding casing.

Reply:
why won't the classic
```
class ApiBase:
    _configured = False

class Satellite:
    _api = ApiBase()

    @property
    def api():
        ....
```

suffice?

Reply:
I don't want to have new helper classes at the module level, this could be confusing for people who might try to import and use them directly. They exist solely to be helper objects that allow attribute injection.